sponsorship_compassion/wizards/migration_r4.py

```diff
@@ -68,12 +68,16 @@ def _create_commitments(self, messages):
         Migrate the old CreateCommitment messages
         """
         logger.info("MIGRATION 8.0.3 ----> CreateCommitment")
+        messages.write({'action_id': self.env.ref(
+            'sponsorship_compassion.create_sponsorship').id})
```

Comment:
You should remove the `unlink` calls on the `_perform_migration` method, otherwise those messages will be deleted.